Philips.CodeAnalysis.SecurityAnalyzers/RegexNeedsTimeoutAnalyzer.cs

```diff
@@ -0,0 +1,76 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.SecurityAnalyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class RegexNeedsTimeoutAnalyzer : DiagnosticAnalyzer
+	{
+		private const string Title = @"RegEx needs a timeout";
+		public const string MessageFormat = @"When constructing a new Regex instance, provide a timeout.";
+		private const string Description = @"When constructing a new Regex instance, provide a timeout (or `RegexOptions.NonBacktracking` in .NET 7 and higher) as this can facilitate denial-of-serice attacks.";
+		private const string Category = Categories.Security;
+		
+		public static readonly DiagnosticDescriptor Rule = new(
+			Helper.ToDiagnosticId(DiagnosticId.RegexNeedsTimeout), 
+			Title, MessageFormat, Category, 
+			DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterSyntaxNodeAction(Analyze, SyntaxKind.ObjectCreationExpression);
```

Comment:
You can try using the new pattern if desired. Or, save for next time.

Reply:
Leave this for a next time